aws/runLambda.sh

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env bash
+
+[[ -n "${GENERATION_DEBUG}" ]] && set ${GENERATION_DEBUG}
+trap '. ${GENERATION_DIR}/cleanupContext.sh; exit ${RESULT:-1}' EXIT SIGHUP SIGINT SIGTERM
+. "${GENERATION_DIR}/common.sh"
+
+
+#Defaults 
+INCLUDE_LOG_TAIL_DEFAULT="false"
+
+tmpdir="$(getTempDir "cote_inf_XXX")"
+
+function usage() {
+    cat <<EOF
+
+Run an AWS Lambda Function
+
+Usage: $(basename $0) -u DEPLOYMENT_UNIT -i INPUT_PAYLOAD -l INCLUDE_LOG_TAIL
+
+where
+
+    -h                  shows this text
+(m) -u DEPLOYMENT_UNIT  is the lambda deployment unit you want to execute
+(o) -i INPUT_PAYLOAD    is the json based payload you want to run the lambda with
+(o) -l INCLUDE_LOG_TAIL include the last 4kb of the execution log
+
+(m) mandatory, (o) optional, (d) deprecated
+
+DEFAULTS:
+
+
+NOTES:
+
+
+EOF
+    exit
+}
+
+
+function options() { 
+
+    # Parse options
+    while getopts ":hi:lu:" opt; do
+        case $opt in
+            h)
+                usage
+                ;;
+            i)
+                INPUT_PAYLOAD="${OPTARG}"
+                ;;
+            l)
+                INCLUDE_LOG_TAIL="true"
+                ;;
+            u)
+                DEPLOYMENT_UNIT="${OPTARG}"
+                ;;
+            \?)
+                fatalOption
+                ;;
+            :)
+                fatalOptionArgument
+                ;;
+        esac
+    done
+
+    #Defaults
+    INCLUDE_LOG_TAIL=${INCLUDE_LOG_TAIL:-$INCLUDE_LOG_TAIL_DEFAULT}
+}
+
+function main() {
+
+    options "$@" || return $?
+
+    # Ensure mandatory arguments have been provided
+    [[ -z "${DEPLOYMENT_UNIT}" ]] && fatalMandatory
+
+    # Set up the context
+    . "${GENERATION_DIR}/setContext.sh"
+
+    # Ensure we are in the right place
+    checkInSegmentDirectory
+
+    # Create build blueprint
+    . "${GENERATION_DIR}/createBuildblueprint.sh" -u "${DEPLOYMENT_UNIT}" 
+
+    COT_TEMPLATE_DIR="${PRODUCT_INFRASTRUCTURE_DIR}/cot/${ENVIRONMENT}/${SEGMENT}"
+    BUILD_BLUEPRINT="${COT_TEMPLATE_DIR}/build_blueprint-${DEPLOYMENT_UNIT}-.json"
+    DEPLOYMENT_UNIT_TYPE="$(jq -r '.Type' < "${BUILD_BLUEPRINT}" )"
+
+    LAMBDA_OUTPUT_FILE="${tmpdir}/${DEPLOYMENT_UNIT}_output.txt" 
+
+    if [[ "${DEPLOYMENT_UNIT_TYPE}" -ne "lambda" ]]; then 
+        fatal "Component type is not a lambda function"
+        return 255
+    fi
+
+    if [[ -n "${INPUT_PAYLOAD}" ]]; then 
+        INPUT_PAYLOAD="$( echo "${INPUT_PAYLOAD}" | jq -r '.' )"  
+        if [[ -z "${INPUT_PAYLOAD}" ]]; then 
+            fatal "Invalid input payload - must be in JSON format"
+            return 255
+        fi
+    fi 
+
+    LAMBDA_ARN="$( jq -r '.Occurrence.Occurrences[0].State.Attributes.ARN' < "${BUILD_BLUEPRINT}" )"
```

Comment:
There can be more than one lambda function in a single lambda zip and hence multiple functions for one deployment unit. Is there a need for a function name parameter to delineate if > 1 function? Or am I missing something abut the way the blueprint is generated?

Reply:
This run command uses the occurrence to lookup the function to run rather than the component. So for any give runLambda there would only be one function.

Reply:
All I can change is the deployment unit, so if I want to run the second lambda function defined for a deployment unit with three lambda functions, I can't? Use of the occurrence would just give me the first?

I'm missing something here...

Reply:
Nope, I am. Didn't realise you could have multiple lambdas under the same deployment unit. Thought that the deployment unit was defined at the function level rather than the lambda level.

Reply:
Fixed up with functionId parameter also added https://github.com/codeontap/gen3-automation/pull/169 to align the automation code